File: TypeScript/hello_world.ts
/*
This is a comment
*/

// Hello world

console.log("Hello, TypeScript!")

// Variables

var myString = "This is a string"
// myString = 5 Error
console.log(myString)

let myString2: string = "This is a string"
myString2 = "This is a different string"
myString2 = "6"
console.log(myString2)
console.log(typeof myString2)

let myNumber: number = 7
myNumber = myNumber + 4
console.log(myNumber)
console.log(typeof myNumber)
console.log(myNumber - 1)
console.log(myNumber)

console.log(myString + " " + myNumber)

let myNumber2 = 6.5
console.log(myNumber2)
console.log(typeof myNumber2)

console.log(myNumber + myNumber2)

let myBool: boolean = false
myBool = true
console.log(myBool)
console.log(typeof myBool)

let myUndefined: undefined
// myUndefined = "myUndefined" Error
console.log(myUndefined)

// Constantes

const myConst = "My constant property"
// myConst = "Another value" Error
console.log(myConst)

// Control structures

if (myNumber == 10 && myString == "Hello") {
    console.log("The value is 10")
} else if (myNumber == 11 || myString == "Hello") {
    console.log("The value is 11")
} else {
    console.log("The value is neither 10 nor 11")
}

// Functions

function myFunction(): string {
    return "My function"
}

console.log(myFunction())

function sumTwoNumbers(firstNumber: number, secondNumber: number): number {
    return firstNumber + secondNumber
}

console.log(sumTwoNumbers(5, 10))

// List

let myList: Array<string> = ["Lucas", "Borondo", "@lubocadev"]
console.log(myList)

// Set

let mySet: Set<string> = new Set(["Lucas", "Borondo", "@lubocadev", "Lucas"])
mySet.add("36")
console.log(mySet)

// Mapas

let myMap: Map<string, number> = new Map([["Lucas", 36], ["Borondo", 37]])
myMap.set("petrovs", 40)
console.log(myMap)
console.log(myMap.get("Lucas"))

// Loops

for (const value of myList) {
    console.log(value)
}

let myCounter = 0

while (myCounter < myList.length) {
    console.log(myList[myCounter])
    myCounter++
}

// Classes

class MyClass {
    name: string
    age: number

    constructor(name: string, age: number) {
        this.name = name
        this.age = age
    }
}

let myClass: MyClass = new MyClass("Lucas", 36)
console.log(myClass)
console.log(myClass.name)

// Enum
enum MyEnum {
    DART = "dart",
    PYTHON = "python",
    SWIFT = "swift",
    JAVA = "java",
    KOTLIN = "kotlin",
    JAVASCRIPT = "javascript"
}

const myEnum = MyEnum.DART
console.log(myEnum)

File: Java/HelloWorld.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class HelloWorld {
    public static void main(String[] args) {

        // Hello world

        System.out.println("Hello, Java!");

        // Variables

        String myString = "This is a string";
        myString = "Here I change the value of the string";
        System.out.println(myString);

        Integer myInt = 7;
        myInt = myInt + 4;
        System.out.println(myInt);
        System.out.println(myInt - 1);

        Double myDouble = 6.5;
        System.out.println(myDouble);

        Float myFloat = 6.5f;
        System.out.println(myFloat);

        System.out.println(myDouble + " " + myString + " " + myFloat + " " + myInt);

        Boolean myBoolean = true;
        myBoolean = false;
        System.out.println(myBoolean);

        myBoolean = null;
        System.out.println(myBoolean);

        // Constantes

        final String myConstant = "This is a constant string";

        // Control 

        myFloat = null;
        if (myFloat != null) {
            System.out.println(myFloat + 10);
            System.out.println("My Float is not null");
        } else {
            System.out.println("My Float is null");
        }

        if (myInt == 11) {
            System.out.println("myInt == 11");
        } else if (myInt == 12) {
            System.out.println("myInt == 12");
        } else {
            System.out.println("The value is neither 11 nor 12");
        }

        // Lists

        List<String> myList = new ArrayList();
        myList.add(myString);
        myList.add(myInt.toString());
        System.out.println(myList);

        // Maps

        Map<String, String> myMap = new HashMap();
        myMap.put("string", myString);
        myMap.put("int", myInt.toString());
        System.out.println(myMap);
        System.out.println(myMap.get("int"));

        // Loops

        for (int index = 0; index < myList.size(); index++) {
            System.out.println(myList.get(index));
        }

        // Classes

        HelloWorld myMain = new HelloWorld();
        System.out.println(myMain.myFunction(5, 2));
        System.out.println(myMain.myFunction(3, 126389));
    }

    // Functions

    public int myFunction(int myFirstNumber, int mySecondNumber) {
        return myFirstNumber + mySecondNumber;
    }

}